### ISSUE TITLE
pulumi_azure.compute.LinuxVirtualMachineScaleSet - adding tags causes unhandled errors in runtime/rpc.py

### ISSUE THREAD

**Issue by kuba-lilz** [pulumi/pulumi-azure#684]
pulumi version: `v2.9.1`
pulumi azure version: `3.19.0`
python: `3.8.5`

With pulumi azure version 3.18.1 or 3.19.0 using tags in `pulumi_azure.compute.LinuxVirtualMachineScaleSet` causes errors

```
      File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/rpc.py", line 436, in <dictcomp>
        translate_output_properties(v, output_transformer, get_type(k))
      File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/rpc.py", line 451, in translate_output_properties
        raise AssertionError(f"Unexpected type. Expected 'list' got '{typ}'")
    AssertionError: Unexpected type. Expected 'list' got '<class 'str'>'
    error: an unhandled error occurred: Program exited with non-zero exit code: 1
```

Running in verbose mode (`pulumi up --logtostderr -v=9`:
```
I0831 11:36:27.896005   16169 eventsink.go:62] eventSink::Debug(<{%reset%}>incoming output property translated: provisionVmAgent -> provision_vm_agent<{%reset%}>)
I0831 11:36:27.896152   16169 eventsink.go:59] incoming output value translated: True -> True
I0831 11:36:27.896158   16169 eventsink.go:62] eventSink::Debug(<{%reset%}>incoming output value translated: True -> True<{%reset%}>)
I0831 11:36:27.896368   16169 eventsink.go:59] RPC failed with exception:
I0831 11:36:27.896373   16169 eventsink.go:62] eventSink::Debug(<{%reset%}>RPC failed with exception:<{%reset%}>)
I0831 11:36:27.896901   16169 eventsink.go:59] Traceback (most recent call last):
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/rpc_manager.py", line 67, in rpc_wrapper
    result = await rpc
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/resource.py", line 426, in do_register
    await rpc.resolve_outputs(res, resolver.serialized_props, resp.object, resolvers)
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/rpc.py", line 487, in resolve_outputs
    translated_value = translate_output_properties(value, res.translate_output_property, types.get(key))
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/rpc.py", line 434, in translate_output_properties
    return {
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/rpc.py", line 436, in <dictcomp>
    translate_output_properties(v, output_transformer, get_type(k))
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/rpc.py", line 451, in translate_output_properties
    raise AssertionError(f"Unexpected type. Expected 'list' got '{typ}'")
AssertionError: Unexpected type. Expected 'list' got '<class 'str'>'
I0831 11:36:27.896915   16169 eventsink.go:62] eventSink::Debug(<{%reset%}>Traceback (most recent call last):
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/rpc_manager.py", line 67, in rpc_wrapper
    result = await rpc
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/resource.py", line 426, in do_register
    await rpc.resolve_outputs(res, resolver.serialized_props, resp.object, resolvers)
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/rpc.py", line 487, in resolve_outputs
    translated_value = translate_output_properties(value, res.translate_output_property, types.get(key))
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/rpc.py", line 434, in translate_output_properties
    return {
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/rpc.py", line 436, in <dictcomp>
    translate_output_properties(v, output_transformer, get_type(k))
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/rpc.py", line 451, in translate_output_properties
    raise AssertionError(f"Unexpected type. Expected 'list' got '{typ}'")
AssertionError: Unexpected type. Expected 'list' got '<class 'str'>'
<{%reset%}>)
I0831 11:36:27.897111   16169 eventsink.go:86] eventSink::Error(<{%reset%}>Program failed with an unhandled exception:<{%reset%}>)
I0831 11:36:27.897694   16169 eventsink.go:86] eventSink::Error(<{%reset%}>Traceback (most recent call last):
  File "/home/kuba/.pulumi/bin/pulumi-language-python-exec", line 85, in <module>
    loop.run_until_complete(coro)
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/stack.py", line 83, in run_in_stack
    await run_pulumi_func(lambda: Stack(func))
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/stack.py", line 51, in run_pulumi_func
    await RPC_MANAGER.rpcs.pop()
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/rpc_manager.py", line 67, in rpc_wrapper
    result = await rpc
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/resource.py", line 426, in do_register
    await rpc.resolve_outputs(res, resolver.serialized_props, resp.object, resolvers)
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/rpc.py", line 487, in resolve_outputs
    translated_value = translate_output_properties(value, res.translate_output_property, types.get(key))
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/rpc.py", line 434, in translate_output_properties
    return {
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/rpc.py", line 436, in <dictcomp>
    translate_output_properties(v, output_transformer, get_type(k))
  File "/home/kuba/anaconda3/envs/pulumi/lib/python3.8/site-packages/pulumi/runtime/rpc.py", line 451, in translate_output_properties
    raise AssertionError(f"Unexpected type. Expected 'list' got '{typ}'")
AssertionError: Unexpected type. Expected 'list' got '<class 'str'>'
<{%reset%}>)
I0831 11:36:28.010920   16169 langruntime_plugin.go:163] langhost[python].RunPlan(pwd=/nas_01/Infrastructure/lilzgauge_pulumi,program=.,...,dryrun=true) success: progerr=Program exited with non-zero exit code: 1
I0831 11:36:28.011290   16169 source_eval.go:167] EvalSourceIterator ended with an error: an unhandled error occurred: Program exited with non-zero exit code: 1
I0831 11:36:28.011301   16169 plan_executor.go:219] planExecutor.Execute(...): incoming event (nil? true, &{0xc0018ddc00})
I0831 11:36:28.011313   16169 eventsink.go:86] eventSink::Error(<{%reset%}>an unhandled error occurred: Program exited with non-zero exit code: 1<{%reset%}>)
I0831 11:36:28.011322   16169 step_executor.go:321] StepExecutor worker(-1): StepExecutor.waitForCompletion(): waiting for worker threads to exit
I0831 11:36:28.011326   16169 step_executor.go:321] StepExecutor worker(-2): worker exiting due to cancellation
I0831 11:36:28.011333   16169 step_executor.go:321] StepExecutor worker(-1): StepExecutor.waitForCompletion(): worker threads all exited
I0831 11:36:28.011338   16169 plan_executor.go:254] planExecutor.Execute(...): step executor has completed
I0831 11:36:28.011356   16169 plan_executor.go:131] planExecutor.Execute(...): exiting provider canceller
I0831 11:36:28.013519   16169 provider_plugin.go:466] Provider[azure, 0xc0002b02a0].Configure() failed: err=transport is closing
I0831 11:36:28.015354   16169 ignore.go:44] Explicitly ignoring and discarding error: rpc error: code = Canceled desc = grpc: the client connection is closing
```

You can reproduce the problem with sample from pulumi's documentation:
( https://www.pulumi.com/docs/reference/pkg/azure/compute/linuxvirtualmachinescaleset/#linuxvirtualmachinescalesetnetworkinterfaceipconfiguration )
by just adding simple tags to it:

```python
example_linux_virtual_machine_scale_set = pulumi_azure.compute.LinuxVirtualMachineScaleSet("exampleLinuxVirtualMachineScaleSet",
    resource_group_name=example_resource_group_name,
    sku="Standard_F2",
    instances=1,
    admin_username="adminuser",
    admin_ssh_keys=[{
        "username": "adminuser",
        "publicKey": public_ssh_key,
    }],
    source_image_reference={
        "publisher": "Canonical",
        "offer": "UbuntuServer",
        "sku": "16.04-LTS",
        "version": "latest",
    },
    os_disk={
        "storage_account_type": "Standard_LRS",
        "caching": "ReadWrite",
    },
    network_interfaces=[{
        "name": "example",
        "primary": True,
        "ip_configurations": [{
            "name": "internal",
            "primary": True,
            "subnet_id": sample_subnet.id
        }],
    }],
    tags={"say": "hello"}
)
```

Tags worked as expected with `pulumi azure 3.18.0`

The worst part about this bug was that there is nothing in python stack trace nor verbose output that indicates tags were the problem. It took a fair bit of trial and error to pin this down.

**Comment by kuba-lilz** [comment]
This is a show stopper for us, and I would imagine many other teams.

**Comment by kuba-lilz** [comment]
Correction - I just checked and last pulumi-azure version that works was `3.17.0`.
Tagging is broken already with `3.18.0`.

**Comment by justinvp** [comment]
Hi @kuba-lilz, sorry for the trouble. We added some extra type checks in the Python SDK in v2.9.0, which mistakenly errors when some internal values are returned from the engine during previews. The issue has been addressed in https://github.com/pulumi/pulumi/pull/5251 and should be released soon. In the meantime, you can pin to `pulumi-azure` `3.17.0`. I'll keep this issue open until we've released an updated Python `pulumi` package with the fix.

**Comment by justinvp** [comment]
The fix has been released in version `2.9.2` of the `pulumi` package, which will work with `pulumi-azure` version `3.19.0`.

**Comment by kuba-lilz** [comment]
Great to here that, thank you for quickly addressing the issue!
